plugins/thunderbolt/fu-thunderbolt-device.c

```diff
@@ -321,36 +323,45 @@ fu_thunderbolt_device_setup (FuDevice *device, GError **error)
 			fu_device_add_instance_id (device, domain_id);
 	}
 
+	/* determine if we can update on unplug */
+	if (fu_udev_device_get_sysfs_attr (FU_UDEV_DEVICE (device),
+					     "nvm_authenticate_on_disconnect",
+					     NULL) != NULL) {
+		self->auth_method = "nvm_authenticate_on_disconnect";
+		/* flushes image */
+		fu_device_add_flag (device, FWUPD_DEVICE_FLAG_USABLE_DURING_UPDATE);
+		/* forces the device to write to authenticate on disconnect attribute */
+		fu_device_remove_flag (device, FWUPD_DEVICE_FLAG_SKIPS_RESTART);
+	} else {
+		self->auth_method = "nvm_authenticate";
+	}
+
 	/* success */
 	return TRUE;
 }
 
 static gboolean
-fu_thunderbolt_device_trigger_update (FuDevice *device, GError **error)
+_write_simple_sysfs (gchar *path, const gchar* val, GError **error)
```

Comment:
could we split this out in FuUdevDevice?

Reply:
You mean make a "write simple sysfs" method for FuUdevDevice that we call?

Reply:
I just pushed what I think you mean.